Run `python3 dev/update_changelog.py --prev-version 2.1...` (#12372)
mlflow/mlflow: CHANGELOG.md

# CHANGELOG

> ## 2.14.0 (2024-06-17)
> 
> MLflow 2.14.0 includes several major features and improvements
> 
> Features:
> 
> - [Models] [Transformers] Support input images encoded with b64.encodebytes (#12087, @MadhuM02)
> - [] Unity Catalog integration for deployments server (#12243, @harupy)
> - [Tracking] Support async logging per X seconds (#12324, @chenmoneygithub)
> - [] Introduce log_traces for langchain autologging and deprecate log_inputs_outputs (#12272, @serena-ruan)
> - [Tracking] Provide a way to set urllib's connection number and max size (#12227, @chenmoneygithub)
> - [Projects] Make MLflow project runner supporting submit spark job to databricks runtime >= 13 (#12139, @WeichenXu123)
> - [Artifacts / Deployments / Model Registry / Models / Projects / Recipes / Tracking] Migrate Mlflow API request to databricks sdk authentication way and support OAuth (#12011, @WeichenXu123)
> - [UI] add column description (#11996, @zhouyou9505)
> - [Tracking] Implement Tracing REST APIs (#12091, @B-Step62)
> - [Models] OpenAI flavor autologging (#11749, @gabrielfu)

Reviewer: ```suggestion
## 2.14.0 (2024-06-17)

MLflow 2.14.0 includes several major features and improvements that we're very excited to announce!

### Major features:

- **MLflow Tracing**: Tracing is powerful tool designed to enhance your ability to monitor, analyze, and debug GenAI applications by allowing you to inspect the intermediate outputs generated as your application handles a request. This update comes with an automatic LangChain integration to make it as easy as possible to get started, but we've also implemented high-level fluent APIs, and low-level client APIs for users who want more control over their trace instrumentation. For more information, check out the [guide in our docs](https://mlflow.org/docs/latest/llms/tracing/index.html)!
- **Unity Catalog Integration**: The MLflow Deployments server now has an integration with Unity Catalog, allowing you to leverage registered functions as tools for enhancing your chat application. For more information, check out [this guide](https://mlflow.org/docs/latest/llms/deployments/uc_integration.html)!
- **OpenAI Autologging**: Autologging support has now been added for the OpenAI model flavor. With this feature, MLflow will automatically log a model upon calling the OpenAI API. Each time a request is made, the inputs and outputs will be logged as artifacts. Check out [the guide](https://mlflow.org/docs/latest/llms/openai/guide/index.html#openai-autologging) for more information!

Other Notable Features:

- [Models] [Transformers] Support input images encoded with b64.encodebytes (#12087, @MadhuM02)
- [Tracking] Support async logging per X seconds (#12324, @chenmoneygithub)
- [Tracking] Provide a way to set urllib's connection number and max size (#12227, @chenmoneygithub)
- [Projects] Make MLflow project runner supporting submit spark job to databricks runtime >= 13 (#12139, @WeichenXu123)
- [Artifacts / Deployments / Model Registry / Models / Projects / Recipes / Tracking] Migrate Mlflow API request to databricks sdk authentication way and support OAuth (#12011, @WeichenXu123)
- [UI] add column description (#11996, @zhouyou9505)
- [Tracking] Implement Tracing REST APIs (#12091, @B-Step62)
```

- **OpenAI Autologging**: Autologging support has now been added for the OpenAI model flavor. With this feature, MLflow will automatically log a model upon calling the OpenAI API. Each time a request is made, the inputs and outputs will be logged as artifacts. Check out [the guide](https://mlflow.org/docs/latest/llms/openai/guide/index.html#openai-autologging) for more information!

Other Notable Features:

- [Models] Support input images encoded with b64.encodebytes (#12087, @MadhuM02)
- [Tracking] Support async logging per X seconds (#12324, @chenmoneygithub)
- [Tracking] Provide a way to set urllib's connection number and max size (#12227, @chenmoneygithub)
- [Projects] Make MLflow project runner supporting submit spark job to databricks runtime >= 13 (#12139, @WeichenXu123)
- [UI] Add the "description" column to the runs table (#11996, @zhouyou9505)

Bug fixes:

- [Model Registry] Handle no headers presigned url (#12349, @artjen)

> - [Tracking] Exclude irrelevant tags from resolve_tags() for tracing (#12330, @xq-yin)
> - [Models] Fix docstring order for ChatResponse class and make object field immutable (#12305, @xq-yin)
> - [Databricks] Fix root user checking in get_databricks_nfs_temp_dir and get_databricks_local_temp_dir (#12186, @WeichenXu123)
> - [Tracking] fix _init_server process terminate hang (#12076, @zhouyou9505)
> - [Scoring] Fix MLflow model container and slow test CI failure (#12042, @WeichenXu123)
> 
> Documentation updates:
> 
> - [Docs] Enhance documentation for autologging supported libraries (#12356, @xq-yin)
> - [Tracking] [MLflow-docs] Adding Langchain as a code example and doc string (#12325, @sunishsheth2009)
> - [Tracking] [MLflow-docs] Adding Pyfunc as a code example and doc string (#12336, @sunishsheth2009)
> - [Docs] Add FAQ entry for viewing trace exceptions in Docs (#12309, @BenWilson2)
> - [Docs] Add note about 'fork' vs 'spawn' method when using multiprocessing for parallel runs (#12337, @B-Step62)
> - [Docs] Fix type error in tracing example for function wrapping (#12338, @B-Step62)
> - [Docs] Add example usage of "extract_fields" for mlflow.search_traces in documentation (#12319, @xq-yin)
> - [Docs] Update LangChain Autologging docs (#12306, @B-Step62)
> - [Docs] Add Tracing documentation (#12191, @BenWilson2)
> 
> Small bug fixes and documentation updates:
> 
> #12359, #12308, #12350, #12284, #12345, #12316, #12287, #12303, #12291, #12288, #12265, #12170, #12248, #12263, #12249, #12251, #12239, #12241, #12240, #12235, #12242, #12172, #12215, #12228, #12216, #12164, #12225, #12203, #12181, #12198, #12195, #12192, #12146, #12171, #12163, #12166, #12124, #12106, #12113, #12112, #12074, #12077, #12058, @harupy; #12355, #12326, #12114, #12343, #12328, #12327, #12340, #12286, #12310, #12200, #12209, #12189, #12194, #12201, #12196, #12174, #12107, @serena-ruan; #12364, #12352, #12354, #12353, #12351, #12298, #12297, #12220, #12155, @daniellok-db; #12311, #12357, #12346, #12312, #12339, #12281, #12283, #12282, #12268, #12236, #12247, #12199, #12232, #12233, #12221, #12229, #12207, #12212, #12193, #12167, #12137, #12147, #12148, #12138, #12127, #12065, @B-Step62; #12289, #12253, @xq-yin; #11771, @lababidi; #12280, #12275, @BenWilson2; #12246, #12244, #12211, #12066, #12061, @WeichenXu123; #12278, @sunishsheth2009; #12136, @kriscon-db; #11911, @jessechancy; #12169, @hubertzub-db
> 

Reviewer: ```suggestion
- [Models] Fix docstring order for ChatResponse class and make object field immutable (#12305, @xq-yin)
- [Databricks] Fix root user checking in get_databricks_nfs_temp_dir and get_databricks_local_temp_dir (#12186, @WeichenXu123)
- [Tracking] fix _init_server process terminate hang (#12076, @zhouyou9505)
- [Scoring] Fix MLflow model container and slow test CI failure (#12042, @WeichenXu123)

Documentation updates:

- [Docs] Enhance documentation for autologging supported libraries (#12356, @xq-yin)
- [Tracking, Docs] Adding Langchain as a code example and doc string (#12325, @sunishsheth2009)
- [Tracking, Docs] Adding Pyfunc as a code example and doc string (#12336, @sunishsheth2009)
- [Docs] Add FAQ entry for viewing trace exceptions in Docs (#12309, @BenWilson2)
- [Docs] Add note about 'fork' vs 'spawn' method when using multiprocessing for parallel runs (#12337, @B-Step62)
- [Docs] Fix type error in tracing example for function wrapping (#12338, @B-Step62)
- [Docs] Add example usage of "extract_fields" for mlflow.search_traces in documentation (#12319, @xq-yin)
- [Docs] Update LangChain Autologging docs (#12306, @B-Step62)
- [Docs] Add Tracing documentation (#12191, @BenWilson2)

Small bug fixes and documentation updates:

#12359, #12308, #12350, #12284, #12345, #12316, #12287, #12303, #12291, #12288, #12265, #12170, #12248, #12263, #12249, #12251, #12239, #12241, #12240, #12235, #12242, #12172, #12215, #12228, #12216, #12164, #12225, #12203, #12181, #12198, #12195, #12192, #12146, #12171, #12163, #12166, #12124, #12106, #12113, #12112, #12074, #12077, #12058, @harupy; #12355, #12326, #12114, #12343, #12328, #12327, #12340, #12286, #12310, #12200, #12209, #12189, #12194, #12201, #12196, #12174, #12107, @serena-ruan; #12364, #12352, #12354, #12353, #12351, #12298, #12297, #12220, #12155, @daniellok-db; #12311, #12357, #12346, #12312, #12339, #12281, #12283, #12282, #12268, #12236, #12247, #12199, #12232, #12233, #12221, #12229, #12207, #12212, #12193, #12167, #12137, #12147, #12148, #12138, #12127, #12065, @B-Step62; #12289, #12253, #12330 @xq-yin; #11771, @lababidi; #12280, #12275, @BenWilson2; #12246, #12244, #12211, #12066, #12061, @WeichenXu123; #12278, @sunishsheth2009; #12136, @kriscon-db; #11911, @jessechancy; #12169, @hubertzub-db

```

- **Unity Catalog Integration**: The MLflow Deployments server now has an integration with Unity Catalog, allowing you to leverage registered functions as tools for enhancing your chat application. For more information, check out [this guide](https://mlflow.org/docs/latest/llms/deployments/uc_integration.html)!
- **OpenAI Autologging**: Autologging support has now been added for the OpenAI model flavor. With this feature, MLflow will automatically log a model upon calling the OpenAI API. Each time a request is made, the inputs and outputs will be logged as artifacts. Check out [the guide](https://mlflow.org/docs/latest/llms/openai/guide/index.html#openai-autologging) for more information!

Other Notable Features:

- [Models] Support input images encoded with b64.encodebytes (#12087, @MadhuM02)
- [Tracking] Support async logging per X seconds (#12324, @chenmoneygithub)
- [Tracking] Provide a way to set urllib's connection number and max size (#12227, @chenmoneygithub)
- [Projects] Make MLflow project runner supporting submit spark job to databricks runtime >= 13 (#12139, @WeichenXu123)
- [UI] Add the "description" column to the runs table (#11996, @zhouyou9505)

> - [Tracking] Implement Tracing REST APIs (#12091, @B-Step62)

Reviewer: ```suggestion
```

- **Unity Catalog Integration**: The MLflow Deployments server now has an integration with Unity Catalog, allowing you to leverage registered functions as tools for enhancing your chat application. For more information, check out [this guide](https://mlflow.org/docs/latest/llms/deployments/uc_integration.html)!
- **OpenAI Autologging**: Autologging support has now been added for the OpenAI model flavor. With this feature, MLflow will automatically log a model upon calling the OpenAI API. Each time a request is made, the inputs and outputs will be logged as artifacts. Check out [the guide](https://mlflow.org/docs/latest/llms/openai/guide/index.html#openai-autologging) for more information!

Other Notable Features:

> - [Models] [Transformers] Support input images encoded with b64.encodebytes (#12087, @MadhuM02)

Reviewer 2: Do we need `[Transformers]`?

> - [Models] [Transformers] Support input images encoded with b64.encodebytes (#12087, @MadhuM02)

Reviewer: ```suggestion
- [Models] Support input images encoded with b64.encodebytes (#12087, @MadhuM02)
```

- [Tracking] Support async logging per X seconds (#12324, @chenmoneygithub)
- [Tracking] Provide a way to set urllib's connection number and max size (#12227, @chenmoneygithub)
- [Projects] Make MLflow project runner supporting submit spark job to databricks runtime >= 13 (#12139, @WeichenXu123)

> - [Artifacts / Deployments / Model Registry / Models / Projects / Recipes / Tracking] Migrate Mlflow API request to databricks sdk authentication way and support OAuth (#12011, @WeichenXu123)

Reviewer: ```suggestion
```

> - [Artifacts / Deployments / Model Registry / Models / Projects / Recipes / Tracking] Migrate Mlflow API request to databricks sdk authentication way and support OAuth (#12011, @WeichenXu123)

Reviewer 2: This is reverted, and the new PR is not merged yet https://github.com/mlflow/mlflow/pull/12313 Let's remove this?

> - [UI] add column description (#11996, @zhouyou9505)

Reviewer: ```suggestion
- [UI] Add the "description" column to the runs table (#11996, @zhouyou9505)
```

> - [Tracking] Implement Tracing REST APIs (#12091, @B-Step62)

Reviewer 2: This is also tracing right, we can remove this?

Bug fixes:

- [Model Registry] Handle no headers presigned url (#12349, @artjen)

> - [Tracking] Exclude irrelevant tags from resolve_tags() for tracing (#12330, @xq-yin)

Reviewer 2: I think we can hide this --> move to small bug fixes section, as no user knows this yet

> - [] Fix docstring order for ChatResponse class and make object field immutable (#12305, @xq-yin)

Reviewer: ```suggestion
- [Models] Fix docstring order for ChatResponse class and make object field immutable (#12305, @xq-yin)
```

> - [] Fix root user checking in get_databricks_nfs_temp_dir and get_databricks_local_temp_dir (#12186, @WeichenXu123)

Reviewer: ```suggestion
- [Databricks] Fix root user checking in get_databricks_nfs_temp_dir and get_databricks_local_temp_dir (#12186, @WeichenXu123)
```

- [Tracking] fix _init_server process terminate hang (#12076, @zhouyou9505)
- [Scoring] Fix MLflow model container and slow test CI failure (#12042, @WeichenXu123)

Documentation updates:

- [Docs] Enhance documentation for autologging supported libraries (#12356, @xq-yin)

> - [Tracking] [MLflow-docs] Adding Langchain as a code example and doc string (#12325, @sunishsheth2009)

Reviewer 2: What's the difference between `Docs` and `MLflow-docs`? If no diff I think we can unify
